scripts/adb_devices_abis.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -e
+typeset -a abis
+SCRIPT_NAME=$(basename $0)
+
+# make sure server is running, otherwise sdb will show devices offline
+adb start-server
+
+adb_devices=$(adb devices | tail -n+2)
+
+while IFS= read line; do
+  read -a arr <<< "${line}"
+
+  device_name="${arr[0]}"
+  device_status=${arr[1]}
+
+  echo "${SCRIPT_NAME} DEBUG: device_name=${device_name}" >&2
+  echo "${SCRIPT_NAME} DEBUG: device_status=${device_status}" >&2
+
+  if [[ "${device_status}" != "device" ]]; then
+    continue
+  fi
+
+  device_product_abi=$(adb -s ${device_name} shell -n getprop ro.product.cpu.abi)
+  echo "${SCRIPT_NAME} DEBUG: device_product_abi=${device_product_abi}" >&2
+
+  if [[ ! ${abis[*]} =~ ${device_product_abi} ]]; then
+    abis+=("${device_product_abi}")
+  fi
+  echo
+done <<< "${adb_devices}"
+
+if [[ ! "${abis[*]}" ]]; then
+  echo "${SCRIPT_NAME}: ERROR: no devices found. Check 'adb devices -l' output and share with Infra team if needed." >&2
```

Comment:
Inconsistent formatting with `:` after `${SCRIPT_NAME}`. This is why you need a `debug` function.